File: test90_out/pollen.parts/Led0/Led0.c

/*
 * ======== string literals ========
 */

static const char pollen_parts_Led0__s__fileName[] = "Led0.p";
static const char pollen_parts_Led0__s__0[] = "Led0";

/*
 * ======== function definitions ========
 */

bool pollen_parts_Led0_isOn__E() {
    {
        return(  local_mcu_Pin0_get__E() == true  );
    }
}

void pollen_parts_Led0_toggle__E() {
    if (pollen_parts_Led0_isOn__E()) {
        pollen_parts_Led0_off__E();
    }
    else {
        pollen_parts_Led0_on__E();
    }
}

void pollen_parts_Led0_targetInit__E() {
    local_mcu_Pin0_makeOutput__E();
    pollen_parts_Led0_off__E();
}

void pollen_parts_Led0_off__E() {
    {
        local_mcu_Pin0_clear__E();
    }
}

void pollen_parts_Led0_on__E() {
    {
        local_mcu_Pin0_set__E();
    }
}

File: test90_out/pollen.parts/Led0/Led0.js

/*
 * ======== MODULE pollen.parts.Led0 ========
 */

var $$c = function() {
    const Led0 = {};
    Led0.Led0 = Led0
    Led0.$name = 'pollen.parts.Led0';
    Led0.pollen$used = 0;
    Led0.targetInit = new $$Ref('pollen_parts_Led0_targetInit__E');
    Led0.on = new $$Ref('pollen_parts_Led0_on__E');
    Led0.off = new $$Ref('pollen_parts_Led0_off__E');
    Led0.toggle = new $$Ref('pollen_parts_Led0_toggle__E');
    Led0.isOn = new $$Ref('pollen_parts_Led0_isOn__E');
    Led0.activeLow = false;
    Led0.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Led0.pollen__uses__ = function() {
    }
    Led0.$$privateInit = function() {
        Led0.pin = 0;
    }
    return Led0;
}
var $$u = $$c();
$units['pollen.parts.Led0'] = $$u;
$units.push($$u);

